arangod/Transaction/Methods.cpp

```diff
@@ -2339,7 +2340,9 @@ Future<OperationResult> transaction::Methods::truncateAsync(std::string const& c
   TRI_ASSERT(_state->status() == transaction::Status::RUNNING);
 
   OperationOptions optionsCopy = options;
-  auto cb = [this, collectionName](OperationResult res) {
+  auto cb = [this, collectionName,
+             handler = rest::RestHandler::CURRENT_HANDLER](OperationResult res) {
+    rest::RestHandler::CURRENT_HANDLER = handler;
     events::TruncateCollection(vocbase().name(), collectionName, res.errorNumber());
     return res;
```

Comment:
should the CURRENT_HANDLER be reverted to its original value after the event was logged, i.e. something like?
```suggestion
    auto old = rest::RestHandler::CURRENT_HANDLER;
    rest::RestHandler::CURRENT_HANDLER = handler;
    events::TruncateCollection(vocbase().name(), collectionName, res.errorNumber());
    rest::RestHandler::CURRENT_HANDLER = old;
    return res;
```